core/market/decentralized/src/matcher/matcher.rs

```diff
@@ -46,8 +47,14 @@ pub enum MatcherError {
     DemandError(#[from] DemandError),
     #[error(transparent)]
     OfferError(#[from] OfferError),
-    #[error("Internal error: {0}.")]
-    InternalError(String),
+    #[error(transparent)]
+    ErrorMessage(#[from] ErrorMessage),
+    #[error(transparent)]
+    SubscriptionParseError(#[from] SubscriptionParseError),
+    #[error(transparent)]
+    SubscriptionValidation(#[from] SubscriptionValidationError),
+    #[error(transparent)]
+    DbError(#[from] DbError),
```

Comment:
If db error happens somewhere, than it should be translated to more meaningful error.